Fabric.Authorization.API/scripts/AzureAD-Migration.ps1

```diff
@@ -0,0 +1,192 @@
+
+    $groups = @();
+    try {	
+        $connection.Open()
+        $adapter = New-Object System.Data.sqlclient.sqlDataAdapter $command
+        $groups = New-Object System.Data.DataTable
+        $adapter.Fill($groups) | Out-Null
+        $connection.Close()
+    }
+    catch [System.Data.SqlClient.SqlException] {
+        Write-DosMessage -Level "Fatal" -Message "An error occurred while executing the command to retrieve non-migrated AD groups. Connection String: $($connectionString). Error $($_.Exception)"
+    }    	
+    
+    Write-DosMessage -Level Information -Message "$($groups.Rows.Count) group(s) found for migration"
+    return $groups;
+}
+
+function Get-AzureADGroupBySID {
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $tenantId,
+        [Parameter(Mandatory=$true)]
+        [string] $groupSID
+    )
+
+    $azureADGroups = @()
+    try {
+        # connect to Azure AD
+        Connect-AzureADTenant -tenantId $tenantId
+
+        Write-DosMessage -Level "Information" -Message "Retrieving group $($groupSID) from Azure AD Tenant $($tenantId)..."
+        $azureADGroups = Get-AzureADGroup -Filter "onPremisesSecurityIdentifier eq '$($groupSID)'"
+
+        # disconnect from Azure AD
+        Disconnect-AzureAD
+    }
+    catch {
+        Write-DosMessage -Level "Fatal" -Message "Error while attempting to retrieve Azure AD group $($groupSID): $($_.Exception)"
+    }
+
+    if ($azureADGroups.Count -eq 1) {
+        return $azureADGroups[0]
+    }
+    elseif ($azureADGroups.Count -eq 0) {
+        $errorMsg = "No match found for SID $($groupSID) in Azure AD."
+        Write-DosMessage -Level "Error" -Message $errorMsg
+        throw $errorMsg
+    }
+    else {
+        $errorMsg = "Multiple matches found for group SID $($groupSID)."
+        Write-DosMessage -Level "Error" -Message $errorMsg
+        throw $errorMsg
+    }
+}
+
+function Move-ActiveDirectoryGroupsToAzureAD {
+     param(
+        [Parameter(Mandatory=$true)]
+        [string] $connString
+    )
+
+    Write-DosMessage -Level "Information" -Message "Migrating AD groups to Azure AD..."
+
+    $allowedTenantIds = Get-AzureADTenants -installConfigPath $installConfigPath
+    if($null -eq $allowedTenantIds -or $allowedTenantIds.Count -eq 0) {
+        Write-DosMessage -Level "Error" -Message  "No tenants were found in the install.config"
+        throw
+    }
+
+    # retrieve all groups from Auth DB    
+    try {
+        $results = Get-NonMigratedActiveDirectoryGroups -connectionString $connString
+    }
+    catch {
+        Write-DosMessage -Level "Fatal" -Message "An error occurred while executing the command to retrieve non-migrated AD groups. Connection String: $($connString). Error $($_.Exception)"
+    }
+
+    if ($results.Count -eq 0) {
+        Write-DosMessage -Level "Information" -Message "No groups found to migrate."
+    }
+
+    foreach ($group in $results) {
+        # query AD for SID
+        $adGroupSID = ""
```

Comment:
Maybe comment about the expected format for the SID here to help whoever has to maintain this code get an idea of what the code is expecting.